server-core/src/main/java/com/stnetix/ariaddna/servercore/tools/ServiceTree.java

```diff
@@ -0,0 +1,126 @@
+package com.stnetix.ariaddna.servercore.tools;
+
+
+import com.stnetix.ariaddna.commonutils.exception.AriaddnaException;
+import com.stnetix.ariaddna.commonutils.logger.AriaddnaLogger;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+/**
+ * Created by alexkotov on 27.02.17.
+ */
+public class ServiceTree {
+    private HashSet<ServiceNode> nodes;
+    private ArrayList<ServiceNode> startQueue;
+
+    private static AriaddnaLogger logger = AriaddnaLogger.getLogger(ServiceTree.class);
+
+    public ServiceTree() {
+        nodes = new HashSet<>();
+    }
+
+    public void addService(IService service, IService parent) throws ServiceNotFoundException {
+        ServiceNode node = new ServiceNode(service, parent);
+        if (!nodes.contains(node)) {
+            nodes.add(node);
+        }
+    }
+
+    public void startServer() throws AriaddnaException {
+        startQueue = createStartQueue();
+        for (ServiceNode node : startQueue) {
+            if (!node.getService().isRun()) {
+                Thread thread = new Thread(node.getService());
+                thread.start();
```

Comment:
It's bad practice run uncontrol tread. Please use PoolExecuter and save references to runnig thread.